Automattic/wp-calypso
# State: Cleanup unused publicize connection request state

#### Changes proposed in this Pull Request

Currently, we track publicize single connection request status, however that's not used anywhere since its creation in #9273.  The `isFetchingConnection` selector that is the sole consumer of that state was never used. 

So this PR cleans this unused state up.

#### Testing instructions

* Verify Calypso builds without any errors.
* Verify all tests still pass: `yarn run test-client client/state/sharing/publicize`
